packages/jest-snapshot/src/index.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+const SnapshotFile = require('./SnapshotFile');
+
+const patchAttr = (attr, state) => {
+  attr.onStart = function(onStart) {
+    return function(context) {
+      const specRunning = context.getFullName();
+      let index = 0;
+      Object.defineProperty(state.specsNextCallCounter, specRunning, {
```

Comment:
I'd really prefer not to use getters here. They are really hard to follow and understand. Can we just add a function increment it? Like

```js
const specName = context.getFullName();
state.getCounter(specName);
state.incrementCounter(specName);
```

I think those names are a bit clearer and if you add functions that modify the state internally, it centralizes all the mutation logic into one module.